pkg/okteto/kubetoken_client.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package okteto
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+	oktetoHttp "github.com/okteto/okteto/pkg/http"
+	"golang.org/x/oauth2"
+)
+
+const kubetokenPath = "auth/kubetoken"
+
+type KubeTokenClient struct {
+	httpClient *http.Client
+	url        string
+}
+
+func NewKubeTokenClient() (*KubeTokenClient, error) {
```

Comment:
Could we add tests for this function? It seems that there are so many things that can go wrong